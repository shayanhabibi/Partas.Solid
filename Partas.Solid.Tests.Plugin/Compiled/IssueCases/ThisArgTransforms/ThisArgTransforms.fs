module Partas.Solid.Tests.IssueCases.ThisArgTransforms

open Partas.Solid.Tests.IssueCases.ThisArgTransformsTypes
open Partas.Solid
open Fable.Core
open Fable.Core.JsInterop
open Fable.Core.JS

[<Erase>]
type PartasTable<'Data>() =
    interface RegularNode

    [<Erase>]
    member val dataStack: DataStack<'Data> = undefined with get, set

    [<SolidTypeComponent>]
    member props.__ =

        let options = TableOptions<'Data>().data (fun () -> props.dataStack.data)
        div ()
