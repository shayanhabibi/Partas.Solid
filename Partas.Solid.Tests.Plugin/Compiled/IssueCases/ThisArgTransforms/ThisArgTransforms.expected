import { splitProps } from "solid-js";
import { DataStack$1__get_data } from "./ThisArgTransformsTypes.fs.jsx";

export function PartasTable(props) {
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, ["dataStack"]);
    let options;
    const this$_1 = {};
    Object.defineProperty(this$_1, "data", {
        get: () => DataStack$1__get_data(PARTAS_LOCAL.dataStack),
    });
    options = this$_1;
    return <div />;
}
